packages/server-wallet/src/protocols/challenge-submitter.ts

```diff
@@ -65,6 +66,8 @@ export class ChallengeSubmitter {
 
       await this.chainService.challenge([signedState], channel.signingWallet.privateKey);
 
+      // TODO: Store the actual finalizesAt instead of 0
```

Comment:
What is the implication of storing `0` instead of the real number?

Reply:
> So the chain service should equally be able to provide the other data that's required, right?
Yes, the `chain-service` could search the events on-chain and find the state info from a `ChallengeRegistered` event before calling `pushOutcomeAndTransferAll`

Although having to scan through the events seems kinda clunky? Uthink listening for `ChallengeRegistered` and storing the challenge state in the database is more straightforward. 

> Is there a reason not to do that now?
Just avoiding extra effort since it's not needed for `challenging-v0`. Although it would be pretty straightforward to do that.

Reply:
Hmm, this might be dangerous.